src/main/java/com/minecolonies/coremod/network/messages/PermissionsMessage.java

```diff
@@ -537,14 +592,78 @@ public void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogicalS
-            if ((permissionsPlayer.getRank() == Rank.HOSTILE && colony.getPermissions().hasPermission(player, Action.EDIT_PERMISSIONS))
-                  || (permissionsPlayer.getRank() != Rank.HOSTILE
+            final Rank hostile = colony.getPermissions().getRankHostile();
+            if ((permissionsPlayer.getRank() == hostile && colony.getPermissions().hasPermission(player, Action.EDIT_PERMISSIONS))
+                  || (permissionsPlayer.getRank() != hostile
                         && colony.getPermissions().hasPermission(player, Action.EDIT_PERMISSIONS)
-                        && colony.getPermissions().getRank(player).ordinal() < colony.getPermissions().getRank(playerID).ordinal())
+                        && permissionsPlayer.getRank() != colony.getPermissions().getRankOwner())
                   || player.getUniqueID().equals(playerID))
             {
                 colony.getPermissions().removePlayer(playerID);
             }
         }
     }
+
+    /**
+     * Message class for removing a rank from a colony
+     */
+    public static class RemoveRank implements IMessage
+    {
+        /**
+         * the colony ID
+         */
+        private int colonyId;
+        /**
+         * the rank ID
+         */
+        private int rankId;
+        /**
+         * the dimension of the colony
+         */
+        private RegistryKey<World> dimension;
+
+        /**
+         * Empty public constructor
+         */
+        public RemoveRank()
+        {
+            super();
+        }
+
+        /**
+         * Constructor for removing a rank from a colony
+         * @param colony the colony to remove the rank from
+         * @param rank the rank to remove
+         */
+        public RemoveRank(@NotNull final IColonyView colony, @NotNull final Rank rank)
+        {
+            super();
+            colonyId = colony.getID();
+            rankId = rank.getId();
+            dimension = colony.getDimension();
+        }
+
+        public void toBytes(@NotNull final PacketBuffer buf)
+        {
+            buf.writeInt(colonyId);
+            buf.writeInt(rankId);
+            buf.writeString(dimension.getLocation().toString());
+        }
+
+        public void fromBytes(@NotNull final PacketBuffer buf)
+        {
+            colonyId = buf.readInt();
+            rankId = buf.readInt();
+            dimension = RegistryKey.getOrCreateKey(Registry.WORLD_KEY, new ResourceLocation(buf.readString(32767)));
+        }
+
+        public void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogicalServer)
+        {
+            final IColony colony = IColonyManager.getInstance().getColonyByDimension(colonyId, dimension);
+            if (colony != null && colony.getPermissions().hasPermission(ctxIn.getSender(), Action.EDIT_PERMISSIONS))
+            {
+                colony.getPermissions().removeRank(colony.getPermissions().getRanks().get(rankId));
```

Comment:
we gotta make sure to move everyone to neutral that has this rank currently

Reply:
happens in Permissions.removeRank()